scipy/integrate/tests/test_ivp.py

```diff
@@ -297,6 +297,7 @@ def test_integration_const_jac():
         assert_allclose(res.sol(res.t), res.y, rtol=1e-14, atol=1e-14)
 
 
+@pytest.mark.timeout(5)
```

Comment:
@drhagen Let me double check if we can accept the additional dependency on `pytest-timeout`. SciPy is usually conservative about this stuff--sorry this is a little tedious!

@rgommers Can we accept this additional test suite dependency? Although a subset of our CI does actually install the requisite library, Travis currently doesn't & it isn't currently needed for local test runs in master for devs / users.

One other thought might be the appropriate time threshold if we ever follow NumPy in dev testing on ARM, which might be slower, but I suppose it could be bumped at a later time if needed.

Reply:
Dependencies for CI are a lot less harmful to add. So I'm not so worried about that here.

The question is more - why is this needed? If the timeout is triggered it looks like the plugin cannot guarantee the rest of the test suite continuing, it may be aborted. I haven't looked at this test, but in principle timeouts are undesirable - why not just trim the execution time as much as possible and then decorate it with `slow` if needed? If there's the possibility of an infinite loop, that indicates a code issue.

Reply:
> why not just trim the execution time as much as possible and then decorate it with slow if needed

This is a test that the stiff ODE solvers are efficient. If the test passes, is passes quickly.. The `timeout` mark was requested because if there is a regression, a possible failure mode is that test takes a very long time to fail, and it was desired that the test quickly fail rather than hang the CI.

> One other thought might be the appropriate time threshold if we ever follow NumPy in dev testing on ARM, which might be slower, but I suppose it could be bumped at a later time if needed.

The timeout number (5 seconds) is an order of magnitude larger than the time it normally takes (0.5 seconds). I would not expect spurious failures even on very slow hardware.

I am personally indifferent to whether or not this is used. I've used `pytest.mark.timeout` before on my own [Python parser](https://github.com/drhagen/parsita/blob/master/tests/test_regex.py#L195) project, so I am comfortable with it. I will update the code to match whatever you guys decide.

Reply:
Okay, that makes sense. I'm fine with either adding `pytest-timeout` or just not worrying about it. I'll leave the decision up to Tyler here.

Reply:
I think I like the tradeoff with `@pytest.mark.slow` instead then if that's not too much trouble.

That way the test is only executed on the full suite and the regression will at least not be silent.
Tradeoff is potential for a bit more dev work connecting a test suite hang to the regression, but perhaps on balance that likely rare scenario is better than the immediate requirement of a new test dependency.